### DISCUSSION TITLE
Problem with Image settings and Automatic1111

### DISCUSSION THREAD

**Discussion by mightykatun** [open-webui/open-webui#7100] [Issues]
## Installation Method

I run open webui on docker, ollama on docker too and automatic1111 as a linux process

## Environment

- **Open WebUI Version:** v0.4.1
- **Ollama (if applicable):** v0.4.2

- **Operating System:** Ubuntu 24.04.1 LTS
- **Browser (if applicable):** Firefox v131.0 (64-bit) Snap for Ubuntu

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

When trying to change image settings with Automatic1111 by pressing the save button, I expected to get a confirmation message.

## Actual Behavior:

The page just freezes and the save button spins forever (see attached image).

## Description

**Bug Summary:**
The save button breaks the page and does not notify of an actual success or failure (same behavior for the connection check button). The settings (for example resolution) seem to be saved though when refreshing the page. My instance of Automatic1111 is reachable by the webui as generating images through the chat works just fine.

## Reproduction Details

**Steps to Reproduce:**
On the web interface, go into admin settings > image settings, modify settings for automatic1111 and press the save button.

## Logs and Screenshots

**Browser Console Logs:**
```plaintext
HTTP/3 422 
date: Wed, 20 Nov 2024 (time redacted)
content-type: application/json
content-length: 141
access-control-allow-origin: https://open_webui.domain.com
access-control-allow-credentials: true
vary: Origin, Origin
x-process-time: 0
cf-cache-status: DYNAMIC
priority: u=0,i=?0
```

**Docker Container Logs:**
```plaintext
INFO:     0:0:0:0 (redacted ip) - "GET /_app/immutable/nodes/12.BtPW7nAa.js.map HTTP/1.0" 304 Not Modified
INFO:     0:0:0:0 (redacted ip) - "POST /images/api/v1/config/update HTTP/1.0" 422 Unprocessable Entity
INFO  [open_webui.apps.images.main] Setting image model to v1-5-pruned-emaonly.safetensors [6ce0161689]
INFO  [open_webui.env] Saving 'IMAGE_GENERATION_MODEL' to the database
INFO  [open_webui.env] Saving 'IMAGE_SIZE' to the database
INFO  [open_webui.env] Saving 'IMAGE_STEPS' to the database
INFO:     0:0:0:0 (redacted ip) - "POST /images/api/v1/image/config/update HTTP/1.0" 200 OK
INFO:     0:0:0:0 (redacted ip) - "GET /_app/immutable/chunks/index.BpuRQpPa.js.map HTTP/1.0" 304 Not Modified
INFO:     0:0:0:0 (redacted ip) - "GET /_app/immutable/chunks/scheduler.DKHHNqU1.js.map HTTP/1.0" 304 Not Modified
INFO:     0:0:0:0 (redacted ip) - "GET /_app/immutable/chunks/index.BG6AZx8_.js.map HTTP/1.0" 304 Not Modified
INFO:     0:0:0:0 (redacted ip) - "GET /_app/immutable/chunks/index.BeZFlfhY.js.map HTTP/1.0" 304 Not Modified
INFO:     0:0:0:0 (redacted ip) - "GET /_app/immutable/chunks/Toaster.svelte_svelte_type_style_lang.B5B2_Ufq.js.map HTTP/1.0" 304 Not Modified
INFO:     0:0:0:0 (redacted ip) - "GET /images/api/v1/models HTTP/1.0" 200 OK
```

**Screenshots/Screen Recordings (if applicable):**
![image](https://github.com/user-attachments/assets/86cf9016-e0dd-4fc6-8ef6-3a9f7e97acfb)
![image](https://github.com/user-attachments/assets/fa9b9e9a-fff9-4852-9d26-ccb1301b3d25)

## Additional Information
200 OK POST and GET in logs when generating the image seen above.

**Comment by tjbck** [comment]
![image](https://github.com/user-attachments/assets/781a7cc1-f5da-4fb1-bb3f-d4d6e0ed0b61)

Unable to reproduce the issue, keep us updated!

**Reply by mightykatun** [reply]
Thank you for the quick answer, it seems to work now!
I purged all config files and reinstalled the container for open webui, reconfigured the connection to automatic1111 and it worked great wth changing anything else, probably some file that got corrupted?
Though after having configured it properly and rebooted my server, the saving of those settings is still a bit unstable and sometimes takes a very long time / webpage freezes unexpectedly (observed the same behavior on multiple devices and browsers).
Nothing that a page refresh doesn't fix but might be something worth investigating if it comes up for other people.

**Comment by 2o1o0** [comment]
Hello,

I currently have this exact issue. I was running 0.4.4 and upgraded to 0.4.7 to see if it solves it self but it doesnt. Its currently deployed with the community helm chart and most settings are enforced via env var. Openwebui is running on a small cluster so automatic1111 itself is running on my gaming gear and I hit it remotly. 

When I try to change an Images settings in the ui it just freeze the current "session" and I need to hit F5 to get control back. I tried to disable all Images settings in env var and restart, but I still get the same behaviour.

Do you want any details for investigation or should I just wipe my install? 

In the second case I was wondering if feat like knowledge and tools could have their data set at initial startup automatically.

Thanks!

**Reply by mightykatun** [reply]
The same problem is still present for me, I think it's just an UI problem though as settings (changed resolution a couple times) seem to have updated once I reload the page.

I did wipe my install, and curiously enough, the first time I set up the connection, it worked perfectly, said something like "connection verified" in green and then saved perfectly. The freezing issue persisted as soon as I tried to change anything however.

I don't mind as much as I can still change settings if I try a couple times as mentionned above.

I'd love to know if anyone else has the same problem.

**Reply by 2o1o0** [reply]
So behavior is slightly different for me as the change is never saved. Despited having the env var enforcing images to be enabled by default, it is not, but all over settings fields are prepopulated properly.

If I change any settings in the images tab, its frozen and I need to refresh. I get the following error in logs : 

`"POST /images/api/v1/config/update HTTP/1.1" 422 Unprocessable Entity`

![image](https://github.com/user-attachments/assets/df508540-222a-4cbb-9633-00f8f7850c35)

**Reply by raydoomed** [reply]
As long as you modify or restart automatic1111, it will freeze

**Reply by 2o1o0** [reply]
I've tried to reenable it from env vars without success. I also tried to push some empty strings with env vars for the url and model, but it doesnt register my settings anymore. In startup logs I see that they are feeded from database instead of env vars?

` [open_webui.env] 'AUTOMATIC1111_BASE_URL' loaded from the latest database entry`

I've enable debug logging but it doesnt give any extra hints in the logs.

**Reply by raydoomed** [reply]
![mmexport1733681753213.png](https://github.com/user-attachments/assets/6206c1f6-0490-4369-881c-96525b49fdc6)

![mmexport1733681147793.png](https://github.com/user-attachments/assets/5d113fae-2a12-4e9b-ab1f-48e5295b4da9)

I can't solve it until now.

**Reply by mightykatun** [reply]
I tried again multiple times, sometimes saving just doesn't do anything, and when it does do something, I get very similar errors in the console

**Reply by mightykatun** [reply]
It seems to be fixed now, might have been a problem with my reverse proxy and implementation of websockets compatibility.
